client/components/filter-select-control/index.tsx

```diff
@@ -0,0 +1,250 @@
+/**
+ * This is a copy of Gutenberg's CustomSelectControl component, found here:
+ * https://github.com/WordPress/gutenberg/tree/7aa042605ff42bb437e650c39132c0aa8eb4ef95/packages/components/src/custom-select-control
+ *
+ * It has been forked from the existing WooPayments copy of this component (client/components/custom-select-control)
+ * to match this specific select input design with an inline label and option hints.
+ */
+
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { Button } from '@wordpress/components';
+import { check, chevronDown, Icon } from '@wordpress/icons';
+import { useCallback } from '@wordpress/element';
+import classNames from 'classnames';
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect, UseSelectState } from 'downshift';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+
+export interface Item {
+	/** The unique key for the item. */
+	key: string;
+	/** The display name of the item. */
+	name?: string;
+	/** Descriptive hint for the item, displayed to the right of the name. */
+	hint?: string;
+	/** Additional class name to apply to the item. */
+	className?: string;
+	/** Additional inline styles to apply to the item. */
+	style?: React.CSSProperties;
+}
+
+export interface Props< ItemType > {
```

Comment:
Curious about the decision to templateise `Props`.

- Do we want to allow different `ItemType`s? 
- What does that allow us to do?
- What does the TypeScript look like if we lock it down to a single `SelectComponentProps`, is that a better fit for our use case/user model?

I just realised that I'm probably commenting on the Gutenberg code, likely not worthwhile revisiting the Gutenberg code.

Reply:
This was carried over from the WooPayments `CustomSelectControl`. Custom item types and components are a requirement for onboarding country selection. I'm leaving this feature to allow onboarding screens to more easily adopt this new component.